meetings/cranelift/2021/cranelift-08-09.md

```diff
@@ -10,6 +10,117 @@
 
 ## Attendees
 
+* Nick Fitzgerald (nf)
+* Chris Fallin (cf)
+* Andrew Brown (abrown)
+* Johnny Birch (jb)
+* Afonso Bordado (abordado)
+* bjorn3 (b3)
+* Benjamin Bouvier (bb)
 
 ## Notes
 
+* (cf) instruction selection pre-RFC
+  * things have gotten complicated enough that a DSL would be nice
+  * "why will this be different from the old DSL we used to have?"
+  * learned things, passed a complexity boundary
+  * would love your comments and discussion on the pre-RFC!
+  * (abrown) read the pre-RFC, it was good, not convinced that we couldn't just
+    add some abstractions to the existing hand-written backend without going
+    full DSL
+    * fwiw, felt the same way about the original old backend, so maybe just
+      biased towards fixing existing stuff
+    * mostly concerned with easily understanding what is going on
+    * depends on what the DSL looks like
+  * (cf) does it depend on the DSL semantics? if it is really clear what the DSL
+    maps down to thats better?
+    * (abrown) the more clear the better
+  * (bb) also interested in refactorings for the existing backend and how far
+    that can take us
+    * with the old backend, we needed better error messages in the DSL and a
+      debugger for the DSL, etc
+    * building that is a lot of work
+  * (abrown) wouldn't mind keeping generated code in-tree if we go DSL route
+    * don't have to search for the proper cargo out directory to inspect
+      generated code
+    * (cf) interesting. the idiomatic rust approach would be to generate in
+      build.rs
+    * (abrown) didn't peepmatic keep generated stuff in tree?
+    * (nf) yes, but mostly so that everyone building cranelift and not touching
+      peepmatic doesn't have to have z3, and anything we start new shouldn't
+      depend on z3, so it should be a non-issue
+    * (b3) rust-analyzer keeps everything in tree
+  * (cf) prototyping one design point in this space, lots of open details,
+    trying to make sense of it myself, will share once it is more formed
+  * (b3) the DSL should be optional
+    * (cf) the existing APIs should be kept, need a gradual transition, see the
+      horizontal and vertical integration stuff in the pre-RFC
+* (cf) update on regalloc2
+  * being reviewed by Julian Seward from Mozilla and Amaneiu from the Rust
+    Project
+  * Looking to relicense from MPL to Apache + LLVM extension
+    * Some code derived from SpiderMonkey's regalloc, which is MPL
+    * Trying to align with other bytecode alliance projects
+* (jb) more SIMD fuzz bugs coming in
+  * should we have some sort of criteria/guidance for approaches to lowering?
+  * when to use assertions?
+  * when to use move helper functions vs emit a particular instruction directly?
+  * mostly want consistency across the code base
+  * (cf) we should document what invariants we already have, eg:
+    * invariants regalloc.rs expects
+    * missed other example
```

Comment:
The other example was instruction sinking / load-op merging (which has really subtle conditions!).